clients/rospy/src/rospy/timer.py

```diff
@@ -204,7 +207,14 @@ def shutdown(self):
         Stop firing callbacks.
         """
         self._shutdown = True
-        
+        self._period = self.__period
+
+    def startup(self):
```

Comment:
Have you tried to use the timer in this way? I don't think this is the behavior of a Python `Thread` instance. The `run()` is invoked **once** when the thread is being started. As soon as the run function returns the thread is done.

When the `_shutdown` flag is being set the while loop will be stopped and the run function returns and therefore the thread is finished.